sinks/datadog/datadog.go

```diff
@@ -388,21 +392,30 @@ func (dd *DatadogSpanSink) Flush() {
 			Start:    span.StartTimestamp,
 			Duration: span.EndTimestamp - span.StartTimestamp,
 			// TODO don't hardcode
-			Type:    "http",
-			Error:   errorCode,
-			Metrics: metrics,
-			Meta:    tags,
+			Type:  "web",
+			Error: errorCode,
+			Meta:  tags,
 		}
 		serviceCount[span.Service]++
-		finalTraces = append(finalTraces, ddspan)
+		if _, ok := traceMap[span.TraceId]; !ok {
+			traceMap[span.TraceId] = []*DatadogTraceSpan{}
+		}
+		traceMap[span.TraceId] = append(traceMap[span.TraceId], ddspan)
+	}
+	// Smush the spans into a two-dimensional array now that they are grouped by trace id.
+	finalTraces := make([][]*DatadogTraceSpan, len(traceMap))
+	idx := 0
+	for _, val := range traceMap {
+		finalTraces[idx] = val
+		idx++
 	}
 
 	if len(finalTraces) != 0 {
 		// this endpoint is not documented to take an array... but it does
 		// another curious constraint of this endpoint is that it does not
 		// support "Content-Encoding: deflate"
 
-		err := vhttp.PostHelper(context.TODO(), dd.HTTPClient, dd.stats, dd.traceClient, fmt.Sprintf("%s/spans", dd.traceAddress), finalTraces, "flush_traces", false, dd.log)
+		err := vhttp.PostHelper(context.TODO(), dd.HTTPClient, dd.stats, dd.traceClient, http.MethodPut, fmt.Sprintf("%s/v0.3/traces", dd.traceAddress), finalTraces, "flush_traces", false, dd.log)
```

Comment:
man, the more parameters we add to this helper the unhappier I get that it exists... I realize there's not much to be done about this now, but we really should think about refactoring this.

Reply:
I completely agree. In fact, when working on this PR I realized we could switch to using [go-datadog-api](https://github.com/zorkian/go-datadog-api) to get rid of this sink's use of HTTP all together. Not in this PR, but we should do it soon!

Reply:
Oh yeah, that's neat. We should get them to extend their library with contexts, though (otherwise we can't cancel these requests anymore)